### DISCUSSION TITLE
Cross compile Ovenmedia engine for custom board

### DISCUSSION THREAD

**Discussion by deekshithkv** [OvenMediaLabs/OvenMediaEngine#1308] [Help]
Hi is it possible to crosscompile ovenmedia engine with custom linux toolchain. we are planning to build it for a embedded board. If it is possible can anyone give the commands/direction to do so.

**Comment by bchah** [comment]
I have compiled OME in Ubuntu, Fedora, Centos, Rocky Linux - on x86_64 (Intel) and arm64v8 (Apple Silicon, Raspberry Pi 3). In all cases it was a matter of fulfilling the prerequisites, then running cmake.

I would first try running prerequisites.sh and troubleshoot getting the various components of OME to install on your platform. If you need to compile them yourself look to the [Common Installation](https://airensoft.gitbook.io/ovenmediaengine/troubleshooting) instructions.

Do you have any more information about the board / platform you are targeting? Other, smarter people than I might be able to help if you can share some more info 👍

**Comment by deekshithkv** [comment]
Thank you for the reply, here is moew details regarding the board in which we are planning to deploy OvenMediaEngine.
We are using Oclea S5L uSoM for production, S5L processor cores is 64-bit ARM Quad Core Cortex - A53 up to 1.0Ghz.
The camera module generates RTSP stream and we would like this as input to OvenMediaEngine and it outputs webRTC. We have a custom toolchain to build applications/libraries for the board by which we are planning to build OvenMediaEngine. I am not sure how to handle this. Can anyone help how it can be done.

**Reply by bchah** [reply]
That board won't have much in the way of spare resources. Since OME does not currently support any of the hardware in the S5L SoC it would need to do everything on CPU. So I wouldn't expect performance beyond that of a Raspberry Pi or another low-power ARM device, particularly on the decoding.

According to Oclea they include an 'Oclea OS' with each board and the idea is that you customize the OS then deploy your final image for mass production. Oclea OS is almost certainly some flavour of embedded Linux so your first goal in terms of OME should be to see if you can get it running on this OS. If you can link me to an ISO I will happily virtualize it and give it a try too (I like fun games).
